all_test.go

```diff
@@ -115,42 +142,60 @@ func setup1(t *testing.T) *Central {
 		sessionDB = newDummySessionDB()
 		permitDB = newDummyPermitDB()
 		roleDB = newDummyRoleGroupDB()
+		userStore = newDummyUserStore()
 		if *backend_ldap {
-			dummyLdapUserStoreAndAuthentictor := newDummyLdapUserStoreAndAuth()
-			userStore = dummyLdapUserStoreAndAuthentictor
-			authenticator = dummyLdapUserStoreAndAuthentictor
-		} else {
-			dummyUserStoreAndAuthentictor := newDummyUserStoreAndAuth()
-			userStore = dummyUserStoreAndAuthentictor
-			authenticator = dummyUserStoreAndAuthentictor
+			ldapTest = newDummyLdap()
+			ldap = ldapTest
+
+
+			permit := setup1_permit()
+			permitDB.SetPermit(joeUserId, &permit)
+			permitDB.SetPermit(samUserId, &permit)
 		}
 	}
-	central := NewCentral(log.Stdout, authenticator, userStore, permitDB, sessionDB, roleDB)
+	central := NewCentral(log.Stdout, ldap, userStore, permitDB, sessionDB, roleDB)
 
-	joePermit := setup1_joePermit()
+	// We add/merge the users to the userstore, and give the users permits
 	if *backend_ldap {
-		err := central.Merge()
+		if !*backend_postgres {
+			ldapTest.AddLdapUser(joeIdentity, joePwd, "joe@gmail.com", "Firstname", "Lastname", "Mobilenumber")
+			ldapTest.AddLdapUser(jackIdentity, jackPwd, "jack@gmail.com", "Firstname", "Lastname", "Mobilenumber")
+			ldapTest.AddLdapUser(SamIdentity, SamPwd, "Sam@gmail.com", "Firstname", "Lastname", "Mobilenumber")
+			ldapTest.AddLdapUser(iHaveNoPermitIdentity, iHaveNoPermitPwd, "iHaveNoPermit@gmail.com", "Firstname", "Lastname", "Mobilenumber")
+		}
+
+		err := central.StartMergeTicker()
```

Comment:
I don't think it's a good idea to run the merge ticker during most unit tests. It will make tests much less predictable. I would run the merge ticker only during tests that explicitly test the merge ticker (eg TestMergeLdap). Inside that test, you'll want to have the merge ticker running on a much tighter frequency. I would suggest 10000 times a second. Then, stress parts of the system that are likely to be in race conditions with the merge ticker. For example the functions Login and GetAuthenticatorIdentities. This is the kind of test that needs to be run with the race detector turned on. You'll want to run that kind of loop exhaustively for a few seconds (say 10 seconds), all the time doing as many Login and GetAuthenticatorIdentities as the machine can do.